scripts/sanitycheck

```diff
@@ -3021,6 +3024,17 @@ Artificially long but functional example:
          west flash -- --board-id=foobar
         """
     )
+    parser.add_argument(
+        "--west-runner", nargs='?', const=[],
+        help="""Uses the specified west runner instead of default"
+
+        E.g
+         sanitycheck --device-testing --device-serial /dev/ttyACM0 \
+                     --west-runner="pyocd"
+        will translate to
+         west flash -r pyocd
+        """
```

Comment:
Based on the code change above, `--west-runner` seems silently ignored when `--west-flash` is missing. However this example in the help doesn't use `--west-flash`, only `--west-runner`?

If `--west-flash` is really required to use `--west-runner` then instead of dropping it on the floor add a warning or maybe even an error if someone or some buggy script makes the mistake of using the latter without the former.

Reply:
west-flash is indeed required for west-runner.  added a check to ensure this was true and generate an error if not.

Reply:
Late suggestion sorry: another, possibly more user-friendly way would be to automatically set `argparse.west_flash=[]` when `--west-runner` is present and `--west-flash` is missing (and of course say so in the --help). It would save the user some typing. Would it make sense from a `west` point of view? I mean:
- Does `--west-runner` make sense with a default `--west-flash` without any explicit value?
- is there really no way some uneducated user could imagine (and be wrong) that `--west-runner` makes sense without any `--west-flash` at all?

Reply:
As this is a flashing command, I would rather set any defaults if the user sets `--west-runner` without enabling `--west-flash`.  I'm open to suggestions here tho.  That's just my opinion.